Scripts/Runtime/Entities/TaskDriver/TaskSet/Job/JobConfig/AbstractJobConfig.cs

```diff
@@ -174,24 +174,24 @@ public IJobConfig RequestCancelFor(AbstractTaskDriver taskDriver)
         // CONFIGURATION - REQUIRED DATA - GENERIC DATA
         //*************************************************************************************************************
 
-        public IJobConfig RequireGenericDataForRead<TData>(IReadOnlyAccessControlledValue<TData> collection)
+        public IJobConfig RequireGenericDataForRead<TData>(IReadAccessControlledValue<TData> collection)
             where TData : struct
         {
             AddAccessWrapper(new GenericDataReadOnlyAccessWrapper<TData>(collection, Usage.Default));
             return this;
         }
 
-        public IJobConfig RequireGenericDataForWrite<TData>(AccessControlledValue<TData> collection)
+        public IJobConfig RequireGenericDataForWrite<TData>(ISharedWriteAccessControlledValue<TData> collection)
```

Comment:
Minor but we should refactor this function name to be:

```suggestion
        public IJobConfig RequireGenericDataForSharedWrite<TData>(ISharedWriteAccessControlledValue<TData> collection)
```

Reply:
Fixed